src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java

```diff
@@ -82,6 +82,15 @@ public void onCompleted(AbstractBuild build, TaskListener listener) {
         additionalComment,
         duration);
 
+    // Mark PR as Approved or Needs Work
+    StashMarkStatus status = new StashMarkStatus();
+    status.handleStatus(
+        trigger.isApproveOnBuildSuccessful(),
+        trigger.isNeedsWorkOnBuildFailure(),
+        cause.getPullRequestId(),
+        build.getResult(),
+        repository);
```

Comment:
As I said on the other PR, I'm not a fan of utility classes. Please move the code from `StashMarkStatus` here. That would be consistent with the code immediately below. If we need to refactor the code, it's better to identify the natural way of doing it, which is easier when similar code resides in one place.

Reply:
done